src/main/scala/com/yotpo/metorikku/input/readers/kafka/KafkaInput.scala

```diff
@@ -45,13 +48,35 @@ case class KafkaInput(name: String, servers: Seq[String], topic: Option[String],
     val kafkaDataFrame = inputStream.load()
     schemaRegistryUrl match {
       case Some(url) => {
-        val schemaRegistryDeserializer = new SchemaRegistryDeserializer(url, topic.getOrElse(""), schemaSubject)
-        schemaRegistryDeserializer.getDeserializedDataframe(sparkSession, kafkaDataFrame)
+        parseStreamDataFrameUsingSchemaRegistry(kafkaDataFrame, url, topic, topicPattern, registeredSchemaVersion, topicPatternGenericSchemaName)
       }
       case None => kafkaDataFrame
     }
   }
 
+  private def parseStreamDataFrameUsingSchemaRegistry(kafkaDataFrame: DataFrame, schemaRegistryUrl: String, topic: Option[String],
+                                                      topicPattern: Option[String], registeredSchemaVersion: Option[String],
+                                                      topicPatternGenericSchemaName: Option[String]): DataFrame = {
+    val registeredSchemaVersionValue = registeredSchemaVersion.getOrElse("latest")
+    val schemaRegistryConfig = Map(
+      SchemaManager.PARAM_SCHEMA_REGISTRY_URL -> schemaRegistryUrl,
+      SchemaManager.PARAM_VALUE_SCHEMA_NAMING_STRATEGY -> SchemaManager.SchemaStorageNamingStrategies.TOPIC_NAME,
+      SchemaManager.PARAM_VALUE_SCHEMA_ID -> registeredSchemaVersionValue
+    )
+    (topic, topicPattern, topicPatternGenericSchemaName) match {
+      case (Some(topic), None, _) => {
+        val schemaTopicRegistryConfig = schemaRegistryConfig + (SchemaManager.PARAM_SCHEMA_REGISTRY_TOPIC -> topic)
+        kafkaDataFrame.select(za.co.absa.abris.avro.functions.from_confluent_avro(col("value"), schemaTopicRegistryConfig) as "value").select("value.*")
+      }
+      case (None, Some(topicPattern), Some(topicPatternGenericSchemaName)) => {
```

Comment:
Basically all you care about is if topicPatternGenericSchemaName (or schemaSubject) was sent, nothing else matters in this case